src/Components/Pages/SetSystemPurposePage.js

```diff
@@ -33,80 +44,83 @@ const SetSystemPurposePage = ({
   const Placeholder = () => (
     <FormSelectOption label="Not defined" isPlaceholder />
   );
-  return isLoading ? (
-    <Loading />
-  ) : (
-    !isError && (
-      <>
-        <Title headingLevel="h2" className="pf-v5-u-mb-sm">
-          Select system purpose
-        </Title>
-        <Text component={TextVariants.p} className="pf-v5-u-mb-xl">
-          System purpose values are used by the subscriptions service to help
-          filter and identify hosts. Setting values for these attributes is an
-          optional step, but doing so ensures that subscriptions reporting
-          accurately reflects the system. Only those values available to your
-          account are shown.
-        </Text>
-        <Form>
-          <FormGroup
-            label="Role"
-            className="pf-v5-u-mb-sm"
-            fieldId="activation-key-role"
+  if (isLoading) return <Loading />;
+  if (isError) return null;
+  return (
+    <>
+      <Title headingLevel="h2" className="pf-v5-u-mb-sm">
+        {isEditMode ? 'Edit system purpose' : 'Select system purpose'}
```

Comment:
Suggestion: Consider if its okay for this to say "Select system purpose" in both the create and edit modes. If so, we can remove any dependence on knowing the context of the modal from this component